Source/Core/Core/Core.cpp

```diff
@@ -146,7 +154,7 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
I don't think the joinable() part is needed here now. We needed it before because a thread doesn't have an id unless it's joinable, but now the is_cpu_thread is per thread so that doesn't matter.